### DISCUSSION TITLE
Problems with Observation Component

### DISCUSSION THREAD

**Discussion by Benusheart** [nazrulworld/fhir.resources#92] [Q&A] [answered]
Good morning,
I am working with FHIR and Python for the first time. As a test, I would like to create an observation that consists of several individual values. It was therefore obvious for me to work with component. Unfortunately, I cannot pass component, no matter what I do:
```
from fhir.resources.observation import Observation
comp={"component": [
      {
        "code": {
          "coding": [
            {
              "system": "http://loinc.org",
              "code": "39125-0",
              "display": "Width of Wound"
            },
            {
              "system": "http://snomed.info/sct",
              "code": "401239006",
              "display": "Width of wound"
            }
          ]
        },
        "valueQuantity": {
          "value": 4,
          "unit": "cm",
          "system": "http://unitsofmeasure.org"
        }
      },
      {
        "code": {
          "coding": [
            {
              "system": "http://loinc.org",
              "code": "39126-8",
              "display": "Length of Wound"
            },
            {
              "system": "http://snomed.info/sct",
              "code": "401238003",
              "display": "Length of wound"
            }
          ]
        },
        "valueQuantity": {
          "value": 3,
          "unit": "cm",
          "system": "http://unitsofmeasure.org"
        }
      }
    ]}
newObservation=Observation.construct()
newObservation.component=comp["component"]
```

i have the component from https://sciphy-stats.com/post/hl7-fhir-fundamentale-create-an-observation-resource/ 
I know there is an ObservationComponent but that didnt work, too.
What am I doing wrong or what do I have to change? Do you have an example? I would be incredibly grateful for an answer!

Update:  Validation with similar component setup would work. Observation: https://ig.fhir.de/basisprofile-de/0.2.30/Blutdruck.html
```
import fhir.resources.observation as obs
from fhir.resources import construct_fhir_element

k={
    "resourceType":"Observation",
    "meta":{
        "profile": [
            "http://fhir.de/StructureDefinition/observation-de-blutdruck/0.2"
        ]
    },
    "status":"final",
    "code":{
        "coding": [
            {
                "system":"http://loinc.org",
                "code":"85354-9"
            }
        ]
    },
    "subject":{
        "reference":"Patient/12345"
    },
    "component": [
        {
            "code":{
                "coding": [
                    {
                        "system":"http://loinc.org",
                        "code":"8480-6"
                    }
                ]
            },
            "valueQuantity":{
                "value":"143",
                "unit":"mmHg",
                "system":"http://unitsofmeasure.org",
                "code":"mm[Hg]"
            }
        },
        {
            "code":{
                "coding": [
                    {
                        "system":"http://loinc.org",
                        "code":"8462-4"
                    }
                ]
            },
            "valueQuantity":{
                "value":"98",
                "unit":"mmHg",
                "system":"http://unitsofmeasure.org",
                "code":"mm[Hg]"
            }
        }
    ]
}

obsi=construct_fhir_element("Observation",k)
print(obsi.json())
obsi2=obs.Observation.construct()
print(obsi2)
obsi2.component=k["component"]
```
but the assignment does not work :

> ValidationError: 1 validation error for Observation
> __root__ -> status
>   none is not an allowed value (type=type_error.none.not_allowed)
> ---------------------------------------------------------------------------
> ValidationError                           Traceback (most recent call last)
> ~\AppData\Local\Programs\Python\Python39\lib\site-packages\pydantic\main.cp39-win_amd64.pyd in pydantic.main.BaseModel.__setattr__()
> 
> ~\AppData\Local\Programs\Python\Python39\lib\site-packages\fhir\resources\observation.py in validate_required_primitive_elements_1353(cls, values)
>     725         if len(errors) > 0:
> --> 726             raise ValidationError(errors, cls)  # type: ignore
>     727 
> 
> ValidationError: 1 validation error for Observation
> status
>   none is not an allowed value (type=type_error.none.not_allowed)
> 
> During handling of the above exception, another exception occurred:
> ValidationError                           Traceback (most recent call last)
> Untitled-1 in <module>
>      61 obsi2=obs.Observation.construct()
>      62 print(obsi2)
> ---> 63 obsi2.component=k["component"]
> 
> ~\AppData\Local\Programs\Python\Python39\lib\site-packages\pydantic\main.cp39-win_amd64.pyd in pydantic.main.BaseModel.__setattr__()
> 
> ValidationError: 1 validation error for Observation
> __root__ -> status
>   none is not an allowed value (type=type_error.none.not_allowed)

**Comment by nazrulworld** [comment, marked as answer]
I would suggest you, please read here https://github.com/nazrulworld/fhir.resources/issues/56
https://github.com/nazrulworld/fhir.resources/pull/57

Let me know if you have any confusion.

FYI:  Observation.status is a mandatory field.

**Reply by Benusheart** [reply]
Thank you very much, that was my mistake!
